common/httpx/tls.go

```diff
@@ -62,7 +63,9 @@ func (h *HTTPX) ZTLSGrab(r *http.Response) *clients.Response {
 	}
 	// canonical net concatenation
 	host = net.JoinHostPort(hostname, fmt.Sprint(port))
-	tlsConn, err := h.Dialer.DialTLS(r.Request.Context(), "tcp", host)
+	ctx, cancel := context.WithTimeout(context.Background(), h.client2.Timeout)
```

Comment:
Shall we use `h.Options.Timeout`?

Reply:
yes. it's default is 10s same as h.client2.Timeout.